### ISSUE TITLE
Docker pdf ocr not working in 0.2.3

### ISSUE THREAD

**Issue by jaxley** [plamola/ocr-joplin-notes#6]
Running via docker generates a fatal error trying to find the pdfinfo command. The Docker image must be missing some of the required components.

```
docker run  --env-file ./docker-env  --network="host" plamola/ocr-joplin-notes:0.2.3 python -m ocr_joplin_notes.cli --mode=FULL_RUN --tag=ojn_markup_joplin-desktop --add-previews=off --autorotation=off
```
The error:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/pdf2image/pdf2image.py", line 441, in pdfinfo_from_path
    proc = Popen(command, env=env, stdout=PIPE, stderr=PIPE)
  File "/usr/local/lib/python3.9/subprocess.py", line 947, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/local/lib/python3.9/subprocess.py", line 1819, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'pdfinfo'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/cli.py", line 89, in <module>
    sys.exit(main())  # pragma: no cover
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/cli.py", line 79, in main
    res = run_mode(mode, tag)
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/ocr_joplin_notes.py", line 79, in run_mode
    return full_run(tag)
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/ocr_joplin_notes.py", line 91, in full_run
    return perform_on_tagged_notes(perform_ocr_for_note, tag_id)
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/joplin_api.py", line 75, in perform_on_tagged_notes
    usage_function(note.get("id"))
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/ocr_joplin_notes.py", line 146, in perform_ocr_for_note
    result_tag = ocr_resources(note)
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/ocr_joplin_notes.py", line 160, in ocr_resources
    data = ocr_resource(resource, ADD_PREVIEWS and note.markup_language == 2)
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/ocr_joplin_notes.py", line 225, in ocr_resource
    ocr_result = extract_text_from_pdf(full_path, language=LANGUAGE)
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/file_ocr.py", line 80, in extract_text_from_pdf
    extracted_image = pdf_page_as_image(filename, i)
  File "/usr/local/lib/python3.9/site-packages/ocr_joplin_notes/file_ocr.py", line 30, in pdf_page_as_image
    pages = convert_from_path(filename, dpi=600, grayscale=True)
  File "/usr/local/lib/python3.9/site-packages/pdf2image/pdf2image.py", line 97, in convert_from_path
    page_count = pdfinfo_from_path(pdf_path, userpw, poppler_path=poppler_path)["Pages"]
  File "/usr/local/lib/python3.9/site-packages/pdf2image/pdf2image.py", line 467, in pdfinfo_from_path
    raise PDFInfoNotInstalledError(
pdf2image.exceptions.PDFInfoNotInstalledError: Unable to get page count. Is poppler installed and in PATH?
```

**Comment by myfta** [comment]
I'm getting the same error with pdf attachments in the Python version (not Docker). As follows:

```
file: 242-a.pdf.gif
Traceback (most recent call last):
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\ocr_joplin_notes.py", line 221, in ocr_resource
    result = extract_text_from_image(full_path, auto_rotate=AUTOROTATION, language=LANGUAGE)
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\file_ocr.py", line 104, in extract_text_from_image
    rotated_image = rotate_image(filename)
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\file_ocr.py", line 54, in rotate_image
    gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
cv2.error: OpenCV(4.5.1) C:\Users\appveyor\AppData\Local\Temp\1\pip-req-build-kh7iq4w7\opencv\modules\imgproc\src\color.cpp:182: error: (-215:Assertion failed) !_src.empty() in function 'cv::cvtColor'


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\graha\Anaconda3\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\graha\Anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\cli.py", line 89, in <module>
    sys.exit(main())  # pragma: no cover
  File "C:\Users\graha\Anaconda3\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\graha\Anaconda3\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "C:\Users\graha\Anaconda3\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\graha\Anaconda3\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\cli.py", line 79, in main
    res = run_mode(mode, tag)
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\ocr_joplin_notes.py", line 79, in run_mode
    return full_run(tag)
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\ocr_joplin_notes.py", line 91, in full_run
    return perform_on_tagged_notes(perform_ocr_for_note, tag_id)
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\joplin_api.py", line 75, in perform_on_tagged_notes
    usage_function(note.get("id"))
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\ocr_joplin_notes.py", line 146, in perform_ocr_for_note
    result_tag = ocr_resources(note)
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\ocr_joplin_notes.py", line 160, in ocr_resources
    data = ocr_resource(resource, ADD_PREVIEWS and note.markup_language == 2)
  File "C:\Users\graha\Anaconda3\lib\site-packages\ocr_joplin_notes\ocr_joplin_notes.py", line 235, in ocr_resource
    os.remove(full_path)
PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\graha\\AppData\\Local\\Temp\\tmpbvdiik62'
```

**Comment by plamola** [comment]
A couple of fixes have been applied to the code base. 
* Poppler has been added to the docker image
* Closing the file is now handled properly. 

Both fixes are available in version 0.3.3
